pkg/controller/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -1722,19 +1723,42 @@ func prepareController(t *testing.T, initObjs ...runtime.Object) (*NSTemplateSet
 		if err != nil {
 			return err
 		}
-		return fakeClient.Client.Create(ctx, o, opts...)
+		innerFakeClient := test.NewFakeClient(t)
```

Comment:
what's the benefit of such an `innerFakeClient`? Why can't we just rely on the `fakeClient.Client`?

Reply:
no, because I want to use the logic that manages the generation number. Since this instance adds its own mocked function for `Create` and `Update` methods so the logic implemented in the fake client is not used and skipped

Reply:
but I could move the logic out of the `Create` and `Update` methods of the fake client and make it available/usable via an exported function - that would also help you in your PR.

Reply:
PR created: https://github.com/codeready-toolchain/toolchain-common/pull/102
changed & used in: 0ff927b

Reply:
thanks, it's actually easier to understand without having to rely on a second client.